Launcher.py

```diff
@@ -463,6 +461,14 @@ def main(args: Optional[Union[argparse.Namespace, dict]] = None):
                 args['component'] = component
             if not component:
                 logging.warning(f"Could not identify Component responsible for {path}")
+        else:
```

Comment:
this elif tree is really confusing this way. not sure if flipping it is worth it here but I'd definiteliy prefer it